File: 3242-count-elements-with-maximum-frequency/count-elements-with-maximum-frequency.java
class Solution {
    public int maxFrequencyElements(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int num : nums) map.put(num, map.getOrDefault(num, 0) + 1);

        int count = 0;
        int maxfreq = Integer.MIN_VALUE;
        for(int freq : map.values()) maxfreq = Math.max(maxfreq, freq);

        for(int freq : map.values()) if(freq == maxfreq) count += maxfreq;
        return count;
    }
}

File: 3242-count-elements-with-maximum-frequency/count-elements-with-maximum-frequency.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxFrequencyElements = function(nums) {
     const map = new Map()

    for(const num of nums) {
        map.set(num, (map.get(num) || 0) + 1)
    }

    let mx = -1, mxNum = 0

    for(const val of map.values()) {
        if(val === mx) {
            mxNum++
        }

        if(val > mx) {
            mxNum = 1
            mx = val
        }
    }

    return mx * mxNum
};